review_practice/api-server/resolvers/room.js

```diff
@@ -0,0 +1,45 @@
+import { AuthenticationError, UserInputError } from 'apollo-server';
+
+// Validator
+const isAdultValueLessEqualThan0 = searchOptions =>
+    searchOptions && searchOptions.personnel && searchOptions.personnel.adult <= 0;
+
+// Resolvers
+const rooms = async (_, args, context) => {
+    if (!context.user) throw new AuthenticationError('Token is expired or do not exist');
+    const { Room } = context.model;
+    const { searchOptions } = args;
+
+    try {
+        if (isAdultValueLessEqualThan0(searchOptions)) throw new UserInputError('Adult need to minimun 1');
+
+        const results = await Room.findAllByFilter(searchOptions);
+
+        return results.map(result => ({
+            room: result,
+            roomOption: result.RoomOption,
+        }));
+    } catch (error) {
+        throw error;
+    }
```

Comment:
어차피 catch 안에서 그대로 throw를 던지는거라면, 굳이 try/catch로 감쌀 필요는 없을 것 같아요!
```suggestion
    if (isAdultValueLessEqualThan0(searchOptions)) throw new UserInputError('Adult need to minimun 1');

    const results = await Room.findAllByFilter(searchOptions);

    return results.map(result => ({
        room: result,
        roomOption: result.RoomOption,
    }));
```